File: left.pl
#!/usr/bin/perl
#
#  Super simple is the goal here.

my $compare  = 0;

my @input = `cat ledger`;
my $total = 0;

if ( @ARGV ) {
   $compare = $ARGV[0];
}

for ( my $i = scalar(@input) ; $i >= 0 ; $i-- ) {
  my $l = $input[$i];

  #if ( $l =~ /\| pot \|/ ) { next; }

  if ( $l =~ /^  \d{2}/ || $l =~ /^  --/ ) {
     if ( ( $l =~ /\| (PAID|SEND)/ && $total != 0 ) || $i == 0 ) {
        print $l;
        print "\nTotal going out: £".sprintf("%.2f",$total)."\n";
        if ( $compare ) {
           print "\nYou have £".sprintf("%.2f left.\n",$compare - $total);
        }
        exit;
     }
     next if ( $l =~ /\| (PAID|SEND)/ );
     my ($date, $date2, $amount) = split(/\|/, $l);
     $total += $amount;
     print $l;
  }
}

File: left.js
const fs = require('fs');
const readline = require('readline');

let compare = false;
let total = 0.00;

if (process.argv.length > 2) {
    compare = process.argv[2];
}

void (async () => {
    const rl = readline.createInterface({
        input: fs.createReadStream('ledger'),
        crlfDelay: Infinity,
    });

    const lines = [];

    rl.on('line', (l) => {
        lines.unshift(l);
    });

    rl.on('close', () => {
        lines.forEach((l) => {
         if (/^  \d{2}/.test(l) || /^  --/.test(l)) {
               if ((/\| (PAID|SEND)/.test(l) && total != 0)) {
                  console.log(l);
                  console.log(`\nTotal going out: £${total.toFixed(2)}\n`);
                  if (compare) {
                     const diff = compare - total;
                     console.log(`\nYou have £${diff.toFixed(2)} left.\n`);
                  }
                  process.exit();
               }
               if (!/\| (PAID|SEND)/.test(l)) {
                  const spl = l.split('|');
                  total += parseFloat(spl[2]);
                  console.log(l);
               }
         }
        });
    });

})();
